pandas/tests/indexes/common.py

```diff
@@ -682,6 +682,21 @@ def test_hasnans_isnans(self):
                 self.assert_numpy_array_equal(idx._isnan, expected)
                 self.assertTrue(idx.hasnans)
 
+    def test_ensure_copied_data(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            if isinstance(index, (PeriodIndex, RangeIndex, MultiIndex)):
+                # Cannot be instantiated from just _data as first argument
+                continue
+
+            data = index._data.copy()
+            idx_type = index.__class__
+            idx = idx_type(data, copy=True)
+            self.assert_numpy_array_equal(data, idx._data)
```

Comment:
Looking at ``i.values.base`` fails for indexes which have ``values`` "artificially" created (only ``CategoricalIndex`` at the moment I guess). How do you see doing ``i.values[0] = i.values[1]`` and then asserting that ``j.values[0]`` doesn't change?